examples/quickstart/matrix_multiplication.cpp

```diff
@@ -0,0 +1,135 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2021 Dimitra Karatza
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+// Parallel implementation of matrix multiplication
+
+#include <hpx/local/algorithm.hpp>
+#include <hpx/local/execution.hpp>
+#include <hpx/local/init.hpp>
+
+#include <cstddef>
+#include <iostream>
+#include <random>
+#include <vector>
+
+std::mt19937 gen;
+
+///////////////////////////////////////////////////////////////////////////////
+//[mul_print_matrix
+void print_matrix(
+    std::vector<int> M, std::size_t rows, std::size_t cols, const char* message)
+{
+    std::cout << "Matrix " << message << " is:" << std::endl;
+    for (std::size_t i = 0; i < rows; i++)
+    {
+        for (std::size_t j = 0; j < cols; j++)
+            std::cout << M[i * rows + j] << " ";
+        std::cout << "\n";
+    }
+}
+//]
+
+///////////////////////////////////////////////////////////////////////////////
+//[mul_hpx_main
+int hpx_main(hpx::program_options::variables_map& vm)
+{
+    using element_type = int;
+
+    // Define matrix sizes
+    std::size_t rowsA = vm["n"].as<std::size_t>();
+    std::size_t colsA = vm["m"].as<std::size_t>();
+    std::size_t rowsB = colsA;
+    std::size_t colsB = vm["k"].as<std::size_t>();
+    std::size_t rowsR = rowsA;
+    std::size_t colsR = colsB;
+
+    // Initialize matrices A and B
+    std::vector<int> A(rowsA * colsA);
+    std::vector<int> B(rowsB * colsB);
+    std::vector<int> R(rowsR * colsR);
+
+    // Define seed
+    unsigned int seed = std::random_device{}();
+    if (vm.count("seed"))
+        seed = vm["seed"].as<unsigned int>();
+
+    gen.seed(seed);
+    std::cout << "using seed: " << seed << std::endl;
+
+    // Define range of values
+    std::size_t lower = vm["l"].as<std::size_t>();
+    std::size_t upper = vm["u"].as<std::size_t>();
+
+    // Matrices have random values in the range [lower, upper]
+    std::uniform_int_distribution<element_type> dis(lower, upper);
+    auto generator = std::bind(dis, gen);
+    hpx::ranges::generate(A, generator);
+    hpx::ranges::generate(B, generator);
+
+    // Perform matrix multiplication
+    hpx::for_loop(hpx::execution::par, 0, rowsA, [&](auto i) {
+        hpx::for_loop(0, colsB, [&](auto j) {
+            R[i * rowsR + j] = 0;
+            hpx::for_loop(0, rowsB, [&](auto k) {
+                R[i * rowsR + j] += A[i * rowsA + k] * B[k * rowsB + j];
```

Comment:
I think the indexing here is still wrong: https://github.com/STEllAR-GROUP/hpx/runs/4464305113?check_suite_focus=true#step:6:137. Could you have another look at this? I may backtrack on saying that we should keep it simple. A `matrix` wrapper would probably be a good idea after all.